huaweicloud/services/cdn/data_source_huaweicloud_cdn_domains.go

```diff
@@ -0,0 +1,307 @@
+				Computed: true,
+			},
+			"origin_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"banned_reason": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"locked_reason": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": common.TagsComputedSchema(),
+		},
+	}
+	return &sc
+}
+
+func sourceSchema() *schema.Resource {
+	sc := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"origin": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"origin_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"active": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"obs_web_hosting_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+
+	return sc
+}
+
+func datasourceDomainsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	var (
+		getCDNDomainsHttpUrl = "v1.0/cdn/domains"
+		getCDNDomainsProduct = "cdn"
+	)
+	getCDNDomainsClient, err := cfg.NewServiceClient(getCDNDomainsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CDN client: %s", err)
+	}
+
+	getCDNDomainsPath := getCDNDomainsClient.Endpoint + getCDNDomainsHttpUrl + "?page_size=10"
+	epsId := cfg.GetEnterpriseProjectID(d)
+	if epsId != "" {
+		getCDNDomainsPath += fmt.Sprintf("&enterprise_project_id=%v", epsId)
+	}
+	if v, ok := d.GetOk("name"); ok {
+		getCDNDomainsPath += fmt.Sprintf("&domain_name=%v", v)
+	}
+	if v, ok := d.GetOk("type"); ok {
+		getCDNDomainsPath += fmt.Sprintf("&business_type=%v", v)
+	}
+	if v, ok := d.GetOk("domain_status"); ok {
+		getCDNDomainsPath += fmt.Sprintf("&domain_status=%v", v)
+	}
+	if v, ok := d.GetOk("service_area"); ok {
+		getCDNDomainsPath += fmt.Sprintf("&service_area=%v", v)
+	}
+	getCDNDomainsOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	currentTotal := 1
+	rst := make([]interface{}, 0)
+	for {
+		currentPath := fmt.Sprintf("%s&page_number=%v", getCDNDomainsPath, currentTotal)
+		getCDNDomainsResp, err := getCDNDomainsClient.Request("GET", currentPath, &getCDNDomainsOpt)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		getCDNDomainsRespBody, err := utils.FlattenResponse(getCDNDomainsResp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		domains := utils.PathSearch("domains", getCDNDomainsRespBody, make([]interface{}, 0)).([]interface{})
+		rst = append(rst, flattenListDomainsBody(domains)...)
+		if len(domains) == 0 {
+			break
+		}
+		currentTotal++
+	}
+	dataSourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(dataSourceId)
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("domains", filterGetDomainsResponseBodyFlavor(rst, d)),
```

Comment:
Suggest performing filter+flatten here instead of performing flatten in the body of the for loop.